### PR TITLE
[DROOLS-6409] if (true) for conditional named consequence fails with …

### PR DESCRIPTION
**Pull request by tkobayas**
…exec-model

**JIRA**: 
https://issues.redhat.com/browse/DROOLS-6409

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by tkobayas** [comment]
Jenkins retest this

**Comment by tkobayas** [comment]
Jenkins retest this

**Comment by tkobayas** [comment]
Jenkins retest this

**Comment by tkobayas** [comment]
Jenkins retest this

**Comment by tkobayas** [comment]
CDB failure is not related to this PR.
```
[2021-06-15T09:35:00.280Z] [INFO] Google Maps ........................................ FAILURE [  0.603 s]
...
[2021-06-15T09:35:00.280Z] [ERROR] Failed to execute goal on project google-maps-workitem: Could not resolve dependencies for project org.jbpm.contrib:google-maps-workitem:jar:7.56.0-SNAPSHOT: Could not transfer artifact com.google.oauth-client:google-oauth-client-java6:jar:1.23.0 from/to mirror-central (https://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8443/nexus/content/groups/kie-all/): /home/jenkins/.m2/repository/com/google/oauth-client/google-oauth-client-java6/1.23.0/google-oauth-client-java6-1.23.0.jar.part (No such file or directory) -> [Help 1]
[2021-06-15T09:35:00.280Z] org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal on project google-maps-workitem: Could not resolve dependencies for project org.jbpm.contrib:google-maps-workitem:jar:7.56.0-SNAPSHOT: Could not transfer artifact com.google.oauth-client:google-oauth-client-java6:jar:1.23.0 from/to mirror-central (https://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8443/nexus/content/groups/kie-all/): /home/jenkins/.m2/repository/com/google/oauth-client/google-oauth-client-java6/1.23.0/google-oauth-client-java6-1.23.0.jar.part (No such file or directory)
```

**Review by mariofusco — APPROVED** [review]

File: drools-model/drools-model-compiler/src/test/java/org/drools/modelcompiler/NamedConsequencesTest.java [hunk 1]
```diff
@@ -589,4 +589,28 @@ public void testMultipleIfAfterEval() {
...
+        KieSession ksession = getKieSession(str);
+        List<String> result = new ArrayList<>();
+        ksession.setGlobal("result", result);
+
+        ksession.insert(new Person("John", 37));
+        ksession.fireAllRules();
+
+        assertEquals(2, result.size());
+        assertTrue(result.containsAll(asList("main", "t1")));
+    }
```
Comment (pibizza) [inline drools-model/drools-model-compiler/src/test/java/org/drools/modelcompiler/NamedConsequencesTest.java:615]: This is an example of a test that is a corner case, perfect as an integration test and not suited to be an acceptance test.

**Review by pibizza — APPROVED** [review]
